File: Python/5.에라토스테네스의 체/2960.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/2960
 '''

n, k = map(int, input().split())
l = [True] * 1001
cnt = 0

l[1] = False
for i in range(2, n+1) :
    for j in range(i, n+1, i) : 
        if l[j] == True :
            cnt+=1
            if cnt == k :
                print(j)
                exit()
        l[j] = False

File: C++/17. 에라토스테네스의 체/2960.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/2960
*/

#include <iostream>
#include <algorithm>

using namespace std;

bool check[1001];
int n, k;

int main(){
    
    cin >> n >> k;
    fill(check, check+1001, true);
    check[1] = false;
    int cnt = 0;
    
    for(int i=2; i<=n; i++){
        for(int j=i; j<=n; j+=i){
            if(check[j] == true){
                check[j] = false;
                cnt++;
                if(cnt == k){
                    cout << j;
                    return 0;
                }
            }
        }
    }
    
    return 0;
}